addons/mail/static/src/attachments/attachment_model.js

```diff
@@ -138,4 +153,8 @@ export class Attachment {
             }
         );
     }
+
+    get voiceDuration() {
+        return this.isVoice && this.filename.split("-")[0];
```

Comment:
it looks like it's one of the special usage of the custom mimetype through `isVoice` but you could use [`this.audioRef.el.duration`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/duration)

Reply:
This data comes from server side where we don't have `audioRef`. I defined a new model for voice message and moved this data to its model so I don't need to hack file name anymore.